omniduct/caches/base.py

```diff
@@ -307,6 +328,127 @@ def has_key(self, key, namespace=None):
         namespace, key = self._namespace(namespace), self._key(key)
         return self._has_key(namespace, key)
 
+    def get_total_bytecount(self, namespaces=None):
+        """
+        Retrieve the total number of bytes used by the cache (excluding metadata).
+
+        Args:
+            namespaces (list<str,None>): The namespaces to which the bytecount
+                should be restricted.
+
+        Returns:
+            int: The total number of bytes used by the nominated namespaces.
+        """
+        total_bytes = 0
+
+        if namespaces is None:
+            namespaces = self.namespaces
+
+        for namespace in namespaces:
+            for key in self.keys(namespace=namespace):
+                total_bytes += self.get_bytecount(key, namespace=namespace)
+
+        return total_bytes
+
+    def describe(self, namespaces=None):
+        """
+        Return a pandas DataFrame showing all keys and their metadata.
+
+        Args:
+            namespaces (list<str,None>): The namespaces to which the summary
+                should be restricted.
+
+        Returns:
+            pandas.DataFrame: A representation of keys in the cache. Will include
+                at least the following columns: ['bytes', 'namespace', 'key',
+                'created', 'last_accessed']. Any additional metadata for keys
+                will be appended to these columns.
+        """
+        out = []
+
+        if namespaces is None:
+            namespaces = self.namespaces
+
+        for namespace in namespaces:
+            for key in self.keys(namespace=namespace):
+                metadata = self.get_metadata(key, namespace=namespace)
```

Comment:
nit: consider inlining metadata and/or usage

Reply:
Done.